### ISSUE TITLE
Authentication - the code you entered has expired

### ISSUE THREAD

**Issue by bdnstn** [lavinir/hassio-onedrive-backup#264]
New installation. First attempt got error "`something went wrong - the code you entered has expired.`"
I restarted the add-on, which gave me a new code and I tried again and it worked. Maybe I was a little too slow the first time around? I have 2FA enabled on my Microsoft account.
I'm reporting the issue just for the benefit of any others who may encounter the same error.

**Comment by pgrodem** [comment]
Same error here, not on a new installation. A restart of the add-on gave me a new code that worked. I guess this is by design, and a security feature from Microsoft. But it might be an idea to include a short message in the "you are not logged in" banner explaining that you can get a new code by restarting the add-on if the code is expired.

**Comment by Wrotaz** [comment]
@bdnstn Yes, I tried. First, the add-on, then the whole HA. The code did not change. Only downgrading resulted in change. Luckily, I do not have any problems running the previous version (everything works correctly).

**Comment by bdnstn** [comment]
How did you become aware of the failure? Did you get a notification in HA or did you just observe that backups were no longer getting copied to OneDrive? If the latter this is a critical issue.

**Comment by lavinir** [comment]
A new code should be generated each time the addon is restarted (if no token cache exists). If not this is a bug and I'd like to see more details if reproducible. 

@bdnstn - The addon exposes sensors showing whether backups are stale for which you can create automations. Additionally backup errors can also show up as Home Assistant Persistent Notifications if you enabled this in the settings.
@Wrotaz - Happy if you can confirm if indeed you were running stale that you were able to see this.

**Comment by Wrotaz** [comment]
Sorry for the late answer:
@bdnstn I wasn't aware until I wanted to make some significant changes in my Zigbee configuration. First, I checked the status of backup (to onedrive) and noticed the add-on was not working and I had to use the code to login to Microsoft. 

@lavinir I updated the add-on to 2.3.6 and this time the add-on hasn't required me to use the code and login to Microsoft. It stays logged in and the upload to OneDrive is working. However, I observed that the yesterday's backup has not been uploaded to OneDrive (it's being done after I updated the add-on). I will observe the behavior in the following days and report if anything odd happens.

**Comment by lavinir** [comment]
@Wrotaz  - When you say the addon wasn't working, did you mean it wasn't running or because it wasn't authenticated? If the latter, it should surface a persistent Home Assistant notification with the error if persistent notifications are enabled in the settings.

**Comment by Wrotaz** [comment]
@lavinir Yes, I meant it. The add-on was working, however it was not authenticated. I just checked and the persistent notifications were not enabled. I must have missed it during some reinstall.
